lighthouse-core/gather/fetcher.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* global document */
+
+class Fetcher {
```

Comment:
would love to get a file overview explaining why this exists, i.e. we run into CORS problems and browser prevents us from reading the response, so we go to protocol and read it there.

Reply:
ya sure.

btw https://github.com/GoogleChrome/lighthouse/issues/4386 could use this too